accepted/0000-yomi-initial-integration.md

```diff
@@ -0,0 +1,361 @@
+consequences, as the state detect early that the root filesystem
+device is already mounted.
+
+Some minor bugs in Uyuni (bsc#1150586, bsc#115089, bsc#115090,
+bsc#1152673, bsc#1152767) makes this process less direct, but for all
+except bsc#1152673 Yomi is providing a workaround. The remaining bug
+requires some manual intervention before the highstate can be applied
+from Uyuni.
+
+The big problem for this workflow has been pointed by Joachim Werner,
+and is about the default values for the pillar data. IIUC the problem
+is that if Yomi provides some default pillar configuration, the new
+discovered node can wrongly apply a highstate (that AFAIK are applied
+automatically or scheduled by the user) with non-optimal data (disk
+configuration, packages, repositories, etc)
+
+There are two proposals, also from Joe, to address this issue (copied
+verbatim from the original post):
+
+-->-->--
+
+1) Discover first.
+
+- Boot a new node (via PXE, UEFI network boot, ISO)
+
+- Register it in SUSE Manager and populate the hardware data
+
+- Shut it down (or leave it running in "wait mode")
+
+- Apply custom configuration by adding formulas and filling in forms
+  (could also be "meta-templated" by SUSE Manager's "activation keys"
+  that define, channels, admin users, groups)
+
+- Boot if necessary
+
+- Apply highstate, including yomi state
+
+2) Pre-populate the DB:
+
+- Create a new "empty" node in the DB and provide the data needed to
+  match it with a physical system (e.g. MAC address or serial number)
+
+- Apply custom configuration by adding formulas and filling in forms
+  (could also be "meta-templated" by SUSE Manager's "activation keys"
+  that define, channels, admin users, groups)
+
+- Boot node (via PXE, UEFI network boot, ISO)
+
+- Register it in SUSE Manager and populate the hardware data
+
+- Apply highstate, including yomi state
+
+(1) exists as a feature, but currently the "discovery image" isn't
+using Salt yet. We could re-use the exact same image that you're using
+for installation, just stop before applying the yomi state.
+
+(2) also exists for saltboot. But as of today it's just an API call,
+no UI version yet. Trivial to add. What's missing is applying an
+activation key to such a system as a template. As of today you'd have
+to inject the activation key grain on the minion side before
+registering the system to Manager.
+
+--<--<--
+
+## Disabling automatic highstate
+
+If I understand the issue correctly, the fact that the Yomi pillar is
+split into multiple sub-forms makes the application of partially
+defined pillar data more probable that if it is defined in a single,
+big form, without defaults.
+
+To potentially fix this issue, I propose in this RFC to have a grain
+emitted by the just-discovered system that will disable any automatic
+highstate application, if present. Only the highstate for this system
+can be manually applied:
+
+```yaml
+  automatic-highstate: no
+```
+
+## High-level forms?
```

Comment:
This will hopefully be fixed once I finish my work on https://github.com/SUSE/spacewalk/issues/9001 and https://github.com/SUSE/spacewalk/issues/8367

Reply:
Very nice. I think that I understand how this can help. I added a note about this in the RFC.